nomad/structs/structs.go

```diff
@@ -2439,7 +2441,7 @@ const (
 
 type NodeNetworkAddress struct {
 	Family        NodeNetworkAF
-	Alias         string
+	Aliases       []string
```

Comment:
Unfortunately this approach breaks backwards compatibility for nodes that are already saved in raft. Given that we can already store multiple `NodeNetworkAddress` for a node, we can solve this problem by writing one for each alias from the node fingerprint.